pysages/_nn/objectives.py

```diff
@@ -0,0 +1,283 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2020-2021: PySAGES contributors
+# See LICENSE.md and CONTRIBUTORS.md at https://github.com/SSAGESLabs/PySAGES
+
+from jax import value_and_grad, vmap
+from plum import dispatch
+from typing import NamedTuple
+
+from .utils import (
+    Float,
+    number_of_weights,
+    pack,
+    prod,
+    sum_squares,
+    unpack,
+)
+
+import jax.numpy as np
+import numpy as onp
+
+
+# Losses
+class Loss:
+    pass
+
+
+class GradientsLoss(Loss):
+    pass
+
+
+class Sobolev1Loss(Loss):
+    pass
+
+
+class SSE(Loss):
+    pass
+
+
+class GradientsSSE(GradientsLoss):
+    pass
+
+
+class Sobolev1SSE(Sobolev1Loss):
+    pass
+
+
+# Regularizers
+class Regularizer:
+    pass
+
+
+class L2Regularization(Regularizer, NamedTuple):
+    coeff: Float
+
+
+class VarRegularization(Regularizer):
+    pass
+
+
+@dispatch
+def build_cost_function(model, loss: Loss, reg: Regularizer):
+    cost = build_cost_function(loss, reg)
```

Comment:
I don't understand, why this is generated here.
Does it have a side effects?
I don't see where `cost` would be used after.